paddle/fluid/operators/mkldnn/softmax_mkldnn_op.cc

```diff
@@ -34,47 +34,61 @@ using platform::to_void_cast;
 
 class SoftmaxMKLDNNHandler : public platform::MKLDNNHandler {
  public:
-  SoftmaxMKLDNNHandler(const platform::MKLDNNDeviceContext& dev_ctx,
+  SoftmaxMKLDNNHandler(const std::vector<int>& dims,
+                       framework::proto::VarType::Type vtype,
+                       const mkldnn::memory::data_type dtype,
+                       const mkldnn::memory::format fmt,
+                       const platform::MKLDNNDeviceContext& dev_ctx,
+                       mkldnn::engine engine, const std::string& base_key)
+      : platform::MKLDNNHandler(dev_ctx, engine, base_key),
+        dims_(dims),
```

Comment:
Perhaps something as common as dimensions and formats could be moved to the generic platform::MKLDNNHandler class. Such action would require more forethought though. Since some operators require several inputs, perhaps they could be stored in two vectors, or one vector containing dims, format pairs.

Reply:
Interesting thought. I hope direction whether to do it or now will become clear once 
we move further with refactoring.